packages/core/src/index.ts

```diff
@@ -606,21 +606,22 @@ class Hopr extends EventEmitter {
       throw new Error('node is not RUNNING')
     }
 
-    const currentChannels: ChannelEntry[] = (await this.getAllChannels()) ?? []
+    let out = 'Channels obtained:\n'
 
-    verbose(`Channels obtained:`)
-    for (const currentChannel of currentChannels) {
-      verbose(currentChannel.toString())
+    const currentChannels: ChannelEntry[] = []
+
+    for await (const channel of this.getAllChannels()) {
+      out += `${channel.toString()}\n`
+      currentChannels.push(channel)
     }
 
+    // Remove last `\n`
+    verbose(out.substring(0, out.length - 1))
+
     if (currentChannels === undefined) {
       throw new Error('invalid channels retrieved from database')
     }
 
-    for (const channel of currentChannels) {
```

Comment:
Where is this functionality replaced?

Reply:
These were two loops over the same array, now combined into one loop iteration